File: 1328. Break a Palindrome.cpp
class Solution {
public:
    string breakPalindrome(string palindrome)
    {
        if (palindrome.length() == 1)
        {
            return "";
        }
        else
        {
            for (int i = 0; i < palindrome.length()/2; i++)
            {
                if (palindrome[i] != 'a')
                {
                    palindrome[i] = 'a';
                    return palindrome;
                }
            }
            for(int i=palindrome.length()-1;i>=palindrome.length()/2;i--){
                if(palindrome[i]!='b'){
                    palindrome[i]='b';
                    return palindrome;
                }
            }
        }
        return palindrome;
    }
};

File: 1328. Break a Palindrome.go
func breakPalindrome(palindrome string) string {
	if len(palindrome) == 1 {
		return ""
	}
	tmp := []rune(palindrome)
	for i := 0; i < len(palindrome)/2; i++ {
		if tmp[i] != 'a' {
			tmp[i] = 'a'
			return string(tmp)
		}
	}
	for i := len(palindrome) - 1; i >= len(palindrome)/2; i-- {
		if tmp[i] != 'b' {
			tmp[i] = 'b'
			return string(tmp)
		}
	}

	return palindrome
}